src/Core/Application/Common/Specification/SpecificationBuilderExtensions.cs

```diff
@@ -79,29 +81,184 @@ public static ISpecificationBuilder<T> PaginateBy<T>(this ISpecificationBuilder<
         return new OrderedSpecificationBuilder<T>(specificationBuilder.Specification);
     }
 
-    private static void AddSearchPropertyByKeyword<T>(this ISpecificationBuilder<T> specificationBuilder, Expression propertyExpr, ParameterExpression paramExpr, string keyword)
+
+    private static MemberExpression GetMemberExpression(string propertyName, ParameterExpression parameter)
+    {
+        Expression mapProperty = parameter;
+        foreach (string member in propertyName.Split('.'))
+        {
+            mapProperty = Expression.PropertyOrField(mapProperty, member);
+        }
+
+        return (MemberExpression)mapProperty;
+    }
+
+    private static BinaryExpression GetBinaryExpression(MemberExpression memberExpression, ConstantExpression constantExpression, ConstantExpression constantExpressionAux, Filter filter)

```

Comment:
Only filter.Operator is used in this method. So better take in only that operator in stead of the whole filter?

Reply:
fixed